### PR TITLE
security: update tough-cookie dependency to 4.1.3

### LINKED ISSUE #31: Vulnerable version of touch-cookie in use (2.5.0) CVE-2023-26136
### Summary
The version of `touch-cookie` package in use is 2.5.0, which is old version and is vulnerable to [CVE-2023-26136](https://www.cve.org/CVERecord?id=CVE-2023-26136)

This should be updated to 4.1.3 (or higher). The main cypress release should then also be updated to include this new release of request.

https://www.cve.org/CVERecord?id=CVE-2023-26136

Package:
https://github.com/cypress-io/request/blob/master/package.json

### Simplest Example to Reproduce
https://security.snyk.io/vuln/SNYK-JS-TOUGHCOOKIE-5672873

### Possible Solution
https://security.snyk.io/vuln/SNYK-JS-TOUGHCOOKIE-5672873

### PR DESCRIPTION
**Pull request by BreakBB**
## PR Checklist:
- [x] I have run `npm test` locally and all tests are passing.
- [x] I have added/updated tests for any new behavior.
       
- [x] If this is a significant change, an issue has already been created where the problem / solution was discussed: [N/A, or add link to issue here]
       
Fixes #31 and therefore [CVE-2023-26136](https://nvd.nist.gov/vuln/detail/CVE-2023-26136)

## PR Description

As described in #31, the tough-cookie dependency had a prototype pollution issue before v4.1.3. This PR updates the tough-cookie dependency to fix this.

### CODE REVIEW

**Review by LubosK — APPROVED** [review]

**Comment by LubosK** [comment]
Hi @tgriesser , could you or someone from the team take a look at this PR? Thx in andvance

**Comment by tim-bezhashvyly** [comment]
+1

**Comment by alfaproject** [comment]
We've also just got our pipeline audit job stalled by this ):

**Comment by LubosK** [comment]
> We've also just got our pipeline audit job stalled by this ):

You can temporary solved it by adding `tough-cookie` to `resolutions` in package.json. 
e.g.
`"resolutions": {
    "tough-cookie": "^4.1.3"
},`

**Comment by tim-bezhashvyly** [comment]
For npm

```json
"overrides": {
    "tough-cookie": "^4.1.3"
  }
```

works as a temporary fix.

**Review by zach-green-proscia — APPROVED** [review]

**Comment by GoudekettingRM** [comment]
Bump

**Comment by herleraja** [comment]
any update on this PR approval ?

**Comment by jeffsays** [comment]
bump - any chance this gets fixed soon @tgriesser?

**Comment by anthony-b-dev** [comment]
needing this approved and merged in as well please

**Review by vmendez-marklogic — APPROVED** [review]

**Review by herleraja — APPROVED** [review]
LGTM. But it needs approval from the maintainer.

**Comment by MikeMcC399** [comment]
I have added related issues which affect the repo without this PR
- #35
- #36
- #37

```bash
yarn add tough-cookie
```

to update to `^4.1.3` causes an additional failure under Node.js `20` (although it is already failing without the change).

**Comment by nagash77** [comment]
Hi everyone, sorry for the late response.  This PR was not seen by our team.  I am having someone take a look to get this PR ready for merge if possible.  I see some failing checks at the moment.  Thank you for your patience.

File: .github/workflows/yarn.yaml [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
+on: [push, pull_request]
 jobs:
   build:
     name: Install and test with yarn
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
+        with:
+          node-version: "18"
```
Comment (MikeMcC399) [inline .github/workflows/yarn.yaml:11]: To get the tests to pass, you need to use Node.js 16.
- See https://github.com/cypress-io/request/issues/37

File: .github/workflows/yarn.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+on: [push, pull_request]
+jobs:
+  build:
+    name: Install and test with yarn
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "18"
```
Comment (MikeMcC399) [inline .github/workflows/yarn.yaml:11]:
```suggestion
          node-version: 16.16.0
```

This matches Cypress' node-version.

**Comment by G-Rath** [comment]
@nagash77 in case you've not seen it, there's also GHSA-p8p7-x288-28g6 which needs addressing - there is #30 and #28 open for that

**Pull request merged**